src/utils/scopeMatch.js

```diff
@@ -0,0 +1,53 @@
+import {
+  T,
+  F,
+  equals,
+  cond,
+  dropLast,
+  pipe,
+  identity,
+  test,
+  not,
+  compose,
+} from 'ramda';
+import { SCOPES_SEARCH_MODE } from './constants';
+
+/**
+ * A match function targeted on scopes.
+ *
+ * scopeMatch :: String mode -> String selectedScope ->
+ *  (String scope -> Boolean isMatch)
+ * */
+const scopeMatch = (mode, selectedScope) => {
+  const exact = equals(selectedScope);
+  const hasScope = cond([
+    [exact, T],
+    [
+      test(/\\*$/),
+      pipe(dropLast(1), scope => selectedScope.indexOf(scope), equals(0)),
+    ],
+    [T, F],
+  ]);
+
+  return cond([
```

Comment:
I like FP, but I can't read this function. If you have to mix a lot of arrow functions into a `cond`, you're losing the benefit of point-free syntax.

I'd probably recommend mixing some more imperative syntax in here if you can't make this fully point-free.

Reply:
Agreed.